app/src/main/java/de/markusfisch/android/shadereditor/widget/ShaderEditor.java

```diff
@@ -372,17 +372,16 @@ private Editable highlight(Editable e) {
 			}
 
 			if (errorLine > 0) {
-				Matcher m = PATTERN_LINE.matcher(e);
-
-				for (int i = errorLine; i-- > 0 && m.find(); ) {
-					// Because analyzers don't like `for ();` statements.
-				}
-
-				e.setSpan(
-						new BackgroundColorSpan(colorError),
-						m.start(),
-						m.end(),
-						Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
+				post(() -> {
```

Comment:
Why `post()`?

And does this actually highlight the correct line for you? On my devices this highlights the line two lines below the error 🤔 By the way, I just noticed error highlighting is off by one line in the current code too!

This works for me:
```
            if (errorLine > 0) {
                int line = errorLine - 2;
                Layout layout = getLayout();
                e.setSpan(
                        new BackgroundColorSpan(colorError),
                        layout.getLineStart(line),
                        layout.getLineEnd(line),
                        Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
            }
```

Reply:
I think the line is off by 2 and not 1, because of the `#define SHADER_EDITOR` that is not seen by the user. I previously had this code somewhere else after changing other parts and there the `post()` was necessary, but here it totally is not.

Reply:
Using this shader I get the correct line:
```glsl
#ifdef GL_FRAGMENT_PRECISION_HIGH
precision highp float;
#else
precision mediump float;
#endif

uniform vec2 resolution;
uniform vec2 resolution; //  S0022: Symbol 'resolution' redeclared

void main(void) {
  vec2 uv = gl_FragCoord.xy / resolution.xy;

  gl_FragColor = vec4(uv, 1.0, 1.0);
}
```

However using this one, I get the wrong one:
```glsl
#define SHADER_EDITOR 1
#ifdef GL_FRAGMENT_PRECISION_HIGH
precision highp float;
#else
precision mediump float;
#endif

uniform vec2 resolution; //  S0022: Symbol 'resolution' redeclared
uniform vec2 resolution; // Error should happen here

void main(void) {
  vec2 uv = gl_FragCoord.xy / resolution.xy;

  gl_FragColor = vec4(uv, 1.0, 1.0);
}
```

Reply:
Ah, yes, you're right! I had already wondered because I had forgotten the definition `SHADER_EDITOR` 👍